lib/plugins/wallet/cardano/wallet.js

```diff
@@ -0,0 +1,76 @@
+const fs = require('fs')
+const { promisify } = require('util')
+
+const cardanoApi = require('./client')
+const crypto = require('crypto')
+
+const coinUtils = require('../../../coin-utils')
+const options = require('../../../options')
+
+const cryptoRec = coinUtils.getCryptoCurrency('ADA')
+const configPath = coinUtils.configPath(cryptoRec)
+
+function sha256(input) {
+  return crypto
+    .createHash('sha256')
+    .update(input)
+    .digest('base64')
+}
+
+async function getCardanoMnemonic() {
+  return promisify(fs.readFile)(options.cardanoMnemonicPath, 'utf8')
+    .then(text => text.trim())
+    .then(mnemonicText => ({
+      mnemonic: mnemonicText.split(/[ \r\n]+/),
+      hash: sha256(mnemonicText)
+    }))
+}
+
+async function getWalletConfiguration() {
+  return promisify(fs.readFile)(configPath)
+    .then(text => JSON.parse(text))
+    .catch(() => ({}))
+}
+
+async function saveWalletConfiguration(configuration) {
+  return fs.writeFile(configPath, JSON.stringify(configuration))
+}
+
+async function walletId() {
+  let configuration = await getWalletConfiguration()
+  const mnemonic = await getCardanoMnemonic()
+
+  if (
```

Comment:
I am not sure I like
1) that `walletId` might do some magic behind. Maybe `getOrCreateWalletId` would be better?
2) creating walletId from config should probably be separated into its own function

Reply:
:+1: